frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -98,22 +108,62 @@ export class FeatureFlagOverviewDetailsSectionCardComponent {
     return this.dialogService.openDisableFeatureFlagConfirmModel(flagName);
   }
 
-  onMenuButtonItemClick(event: 'Edit' | 'Delete' | 'Duplicate', flag: FeatureFlag) {
-    if (event === 'Delete') {
-      this.dialogService.openDeleteFeatureFlagModal();
-    } else if (event === 'Edit') {
-      this.dialogService.openEditFeatureFlagModal(flag);
-    } else if (event === 'Duplicate') {
-      this.dialogService.openDuplicateFeatureFlagModal(flag);
+  onMenuButtonItemClick(event: FEATURE_FLAG_DETAILS_PAGE_ACTIONS, flag: FeatureFlag) {
+    switch (event) {
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DELETE:
+        this.dialogService.openDeleteFeatureFlagModal();
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EDIT:
+        this.dialogService.openEditFeatureFlagModal(flag);
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DUPLICATE:
+        this.dialogService.openDuplicateFeatureFlagModal(flag);
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.ARCHIVE:
+        console.log('Archive feature flag');
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EXPORT_DESIGN:
+        this.openConfirmExportDesignModal();
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EMAIL_DATA:
+        this.openConfirmEmailDataModal();
+        break;
+      default:
+        console.log('Unknown action');
     }
   }
 
+  openConfirmExportDesignModal() {
+    const confirmMessage = 'feature-flags.export-feature-flag-design.confirmation-text.text';
+    this.dialogService.openExportFeatureFlagDesignModal(confirmMessage)
+      .afterClosed()
+      .subscribe((isExportClicked: boolean) => {
+        if (isExportClicked) {
+          this.featureFlagService.exportFeatureFlagsData([this.featureFlag.id]);
+        }
+      });
+  }
+
+  openConfirmEmailDataModal() {
+    const confirmMessage = 'feature-flags.export-feature-flags-data.confirmation-text.text';
+    const emailConfirmationMessage = "The feature flag will be sent to '" + this.emailId + "'." ;
+    this.dialogService.openEmailFeatureFlagDataModal(confirmMessage, emailConfirmationMessage)
+      .afterClosed()
+      .subscribe((isEmailClicked: boolean) => {
+        if (isEmailClicked) {
+          this.featureFlagService.emailFeatureFlagData(this.featureFlag.id);
+        }
+      });
+  }
+
   onSectionCardExpandChange(isSectionCardExpanded: boolean) {
     this.isSectionCardExpanded = isSectionCardExpanded;
     this.sectionCardExpandChange.emit(this.isSectionCardExpanded);
   }
 
-  ngOnDestroy() {
+  ngOnDestroy(): void {
+    this.flagSub.unsubscribe();
```

Comment:
remove flagSub unsubscribe and mailSub unsubscribe, all subcriptions can be destroyed together.